account_invoice_start_end_dates_tax_change/wizards/account_move_apply_tax_change.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import fields, models
+
+
+# NOTE: method copied from recent Odoo versions (odoo.tools.date_utils)
+def date_subtract(value, *args, **kwargs):
+    """
+    Return the difference between ``value`` and a :class:`relativedelta`.
+
+    :param value: initial date or datetime.
+    :param args: positional args to pass directly to :class:`relativedelta`.
+    :param kwargs: keyword args to pass directly to :class:`relativedelta`.
+    :return: the resulting date/datetime.
+    """
+    return value - relativedelta(*args, **kwargs)
+
+
+class AccountMoveApplyTaxChange(models.TransientModel):
+    _inherit = "account.move.apply.tax.change"
+
+    def _skip_line(self, line):
+        if line.start_date or line.end_date:
+            if line.start_date <= self.tax_change_id.date < line.end_date:
+                return False
+            return True
+        return super(AccountMoveApplyTaxChange, self)._skip_line(line)
+
+    def _change_taxes(self, line):
+        if line.start_date or line.end_date:
+            # Start & end dates defined in the scope of the tax changes
+            if line.start_date <= self.tax_change_id.date < line.end_date:
```

Comment:
```suggestion
        if self._is_recurring_line():
            if self._is_recurring_tax_change_required():
(...)
    def _is_recurring_line(self, line):
        return line.start_date and line.end_date
    
    def _is_recurring_tax_change_required(self, line):
        # Start & end dates defined in the scope of the tax changes
        return line.start_date <= self.tax_change_id.date < line.end_date
```
I guess we can refactor to avoid using the same comparison in this function and the one above

Reply:
done